### ISSUE TITLE
__delete__ in descriptor

### ISSUE THREAD

**Issue by mehdigmira** [microsoft/pyright#6636]
**Describe the bug**
Pyright considers attribute is unknown when calling delete on it, even if specific `__delete__` function is implemented

**Code or Screenshots**

```python
from __future__ import annotations

from collections.abc import Callable
from typing import Any, Generic, TypeVar, overload

ReturnT=TypeVar("ReturnT")

class cached_property(Generic[ReturnT]):

    def __init__(self, func: Callable[..., ReturnT], name: str|None = None, doc: str|None = None):
        self.__name__ = name or func.__name__
        self.__module__ = func.__module__
        self.__doc__ = doc or func.__doc__
        self.func = func

    def __set__(self, obj: Any, value: ReturnT):
        obj.__dict__[self.__name__] = value

    @overload
    def __get__(self, obj: None, owner: type|None=None) -> cached_property[ReturnT]:
        ...

    @overload
    def __get__(self, obj: object, owner: type|None=None) -> ReturnT:
        ...

    def __get__(self, obj: object | None, owner: type|None=None) -> ReturnT | cached_property[ReturnT]:
        if obj is None:
            return self  
        if self.__name__ in obj.__dict__:
            value = obj.__dict__[self.__name__]
        else:
            value = self.func(obj)
            obj.__dict__[self.__name__] = value
        return value

    def __delete__(self, obj: Any):
        del obj.__dict__[self.__name__]


class A:

    @cached_property
    def expensive_function(self) -> str:
        print("running")
        return 'running'


a = A()
a.expensive_function

del a.expensive_function  

a.expensive_function # error: Type of "expensive_function" is unknown
```

**VS Code extension or command-line**
pyright 1.1.338

**Comment by erictraut** [comment]
Thanks, I agree that pyright's current behavior is incorrect here. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.339, which I just published. It will also be included in a future release of pylance.
